precompiles/distribution/integration_test.go

```diff
@@ -1442,6 +1442,74 @@ var _ = Describe("Calling distribution precompile from another contract", func()
 			finalBalance := s.app.BankKeeper.GetBalance(s.ctx, s.address.Bytes(), s.bondDenom)
 			Expect(finalBalance.Amount.GT(initialBalance.Amount)).To(BeTrue(), "expected final balance to be greater than initial balance after claiming rewards")
 		})
+
+		Context("Table driven tests", func() {
+			var (
+				args                   contracts.CallArgs
+				contractInitialBalance = math.NewInt(100)
+			)
+			BeforeEach(func() {
+				args = defaultClaimRewardsArgs.
+					WithMethodName("testClaimRewardsWithTransfer").
+					WithGasPrice(gasPrice)
+
+				// send some funds to the contract
+				err := evmosutil.FundAccountWithBaseDenom(s.ctx, s.app.BankKeeper, contractAddr.Bytes(), contractInitialBalance.Int64())
+				Expect(err).To(BeNil())
+			})
+
+			DescribeTable("claimRewards with transfer to withdrawer", func(tc testCase) {
+				// get the pending rewards to claim
+				qr := distrkeeper.Querier{Keeper: s.app.DistrKeeper}
+				qRes, err := qr.DelegationTotalRewards(s.ctx, &distrtypes.QueryDelegationTotalRewardsRequest{DelegatorAddress: sdk.AccAddress(s.address.Bytes()).String()})
+				Expect(err).To(BeNil())
+				expRewards := qRes.Total.AmountOf(s.bondDenom).TruncateInt()
+
+				claimRewardsArgs := args.WithArgs(
+					s.address, uint32(2), tc.before, tc.after,
+				)
+
+				logCheckArgs := passCheck.
+					WithExpEvents(distribution.EventTypeClaimRewards)
+
+				res, _, err := contracts.CallContractAndCheckLogs(s.ctx, s.app, claimRewardsArgs, logCheckArgs)
+				Expect(err).To(BeNil(), "error while calling the smart contract: %v", err)
+				fees := math.NewIntFromBigInt(gasPrice).MulRaw(res.GasUsed)
+
+				// calculate the transferred amt during the call
+				contractTransferredAmt := math.ZeroInt()
+				for _, transferred := range []bool{tc.before, tc.after} {
+					if transferred {
+						contractTransferredAmt = contractTransferredAmt.AddRaw(15)
```

Comment:
where this 15 comes from?

Reply:
is a hardcoded value to transfer on the smart contract function
```
 function testClaimRewardsWithTransfer(
        address payable _delAddr,
        uint32 _maxRetrieve,
        bool _before,
        bool _after
    ) public {
        if (_before) {
            counter++;
            if (_delAddr != address(this)) {
                (bool sent, ) = _delAddr.call{value: 15}("");
                require(sent, "Failed to send Ether to delegator");
            }
        }
        bool success = distribution.DISTRIBUTION_CONTRACT.claimRewards(
            _delAddr,
            _maxRetrieve
        );
        require(success, "failed to claim rewards");
        if (_after) {
            counter++;
            if (_delAddr != address(this)) {
                (bool sent, ) = _delAddr.call{value: 15}("");
                require(sent, "Failed to send Ether to delegator");
            }
        }
    }
```